File: src/test/java/com/github/spartusch/webquery/WebQueryTest.java
package com.github.spartusch.webquery;

import org.junit.Before;
import org.junit.Test;

import java.nio.charset.StandardCharsets;

import static org.assertj.core.api.Assertions.assertThat;

public class WebQueryTest {

    private WebQuery webQuery;

    @Before
    public void setUp() {
        webQuery = new WebQuery("content", StandardCharsets.UTF_8);
    }

    @Test
    public void test_getContentType() {
        assertThat(webQuery.getContentType()).isEqualTo("text/plain; charset=UTF-8");
    }

    @Test
    public void test_getContentLength() {
        assertThat(webQuery.getContentLength()).isEqualTo(webQuery.getContent().length);
    }

    @Test
    public void test_getContentDisposition_withFilename() {
        assertThat(webQuery.getContentDisposition("fn.iqy")).isEqualTo("attachment; filename=fn.iqy");
    }

    @Test(expected = NullPointerException.class)
    public void test_getContentDisposition_withoutFilename() {
        webQuery.getContentDisposition(null);
    }
}

File: src/test/kotlin/com/github/spartusch/webquery/WebQueryTest.kt
package com.github.spartusch.webquery

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class WebQueryTest {

    private lateinit var cut: WebQuery

    @BeforeEach
    fun setUp() {
        cut = WebQuery("content", Charsets.UTF_8)
    }

    @Test
    fun contentBytes_equalsContent() {
        assertThat(cut.contentBytes.contentEquals("content".toByteArray(Charsets.UTF_8))).isTrue()
    }

    @Test
    fun toString_matchesContent() {
        assertThat(cut.toString()).isEqualTo("content")
    }

    @Test
    fun contentLength() {
        assertThat(cut.contentLength).isEqualTo("content".length)
    }

    @Test
    fun contentType() {
        assertThat(cut.contentType).isEqualTo("text/plain; charset=UTF-8")
    }

    @Test
    fun getContentDisposition() {
        assertThat(WebQuery.getContentDisposition("fn.iqy")).isEqualTo("attachment; filename=fn.iqy")
    }
}
